src/runners/junitRunner/JUnitRunnerResultAnalyzer.ts

```diff
@@ -60,40 +64,26 @@ export class JUnitRunnerResultAnalyzer extends RunnerResultAnalyzer {
             if (!item) {
                 return;
             }
-            if (item.id !== this.currentItem?.id) {
-                this.initializeCache(item);
-            }
-            this.testContext.testRun.started(item);
-
-            const start: number = Date.now();
-            if (this.currentDuration === 0) {
-                this.currentDuration = -start;
-            } else if (this.currentDuration > 0) {
-                // Some test cases may executed multiple times (@RepeatedTest), we need to calculate the time for each execution
-                this.currentDuration -= start;
-            }
+            this.setCurrentState(item, TestResultState.Running, 0);
+            this.setDurationAtStart(this.getCurrentState(item));
```

Comment:
(not all changes to this method are strictly necessary, but I felt it becoming a bit lengthy now, so I extracted some details into separate methods)